frontend/src/pages/NewRunV2.tsx

```diff
@@ -167,6 +199,13 @@ function NewRunV2(props: NewRunV2Props) {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  // Determine if it is cloning (run / recurring run)
+  useEffect(() => {
+    if (apiRun || apiRecurringRun) {
+      setIsClone(true);
+    }
+  }, [apiRun, apiRecurringRun]);
```

Comment:
If we write a function which combine `apiRun` and `apiRunRecurringRun` into one `CloneOrigin` object, which contains the information like `isRecurringRun`. Then we can simplify the main logic in `NewRunV2`.